File: iconex/src/main/java/foundation/icon/iconex/view/ui/mainWallet/component/RefreshLoadingView.java
package foundation.icon.iconex.view.ui.mainWallet.component;

import android.content.Context;
import android.view.LayoutInflater;
import android.widget.FrameLayout;

import androidx.annotation.NonNull;

import foundation.icon.iconex.R;
import foundation.icon.iconex.widgets.RefreshLayout.LoadingHeaderListener;

public class RefreshLoadingView extends FrameLayout implements LoadingHeaderListener {
    public RefreshLoadingView(@NonNull Context context) {
        super(context);
        initView();
    }

    private void initView() {
        LayoutInflater.from(getContext()).inflate(R.layout.layout_pull_to_refreash, this, true);
    }

    @Override
    public void onRefreshBefore(int scrollY, int headerHeight) {

    }

    @Override
    public void onRefreshAfter(int scrollY, int headerHeight) {

    }

    @Override
    public void onRefreshReady(int scrollY, int headerHeight) {

    }

    @Override
    public void onRefreshing(int scrollY, int headerHeight) {

    }

    @Override
    public void onRefreshComplete(int scrollY, int headerHeight, boolean isRefreshSuccess) {

    }

    @Override
    public void onRefreshCancel(int scrollY, int headerHeight) {

    }

    @Override
    public int getRefreshHeight() {
        return 0;
    }
}

File: iconex/src/main/java/foundation/icon/iconex/dev_detail/RefreshLoadingView.kt
package foundation.icon.iconex.dev_detail

import android.content.Context
import android.util.AttributeSet
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.FrameLayout
import foundation.icon.iconex.R
import foundation.icon.iconex.widgets.RefreshLayout.LoadingHeaderListener

class RefreshLoadingView: FrameLayout, LoadingHeaderListener {
    constructor(context: Context) : super(context) {
        initView()
    }
    constructor(context: Context, attrs: AttributeSet?) : super(context, attrs) {
        initView()
    }
    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int) : super(context, attrs, defStyleAttr) {
        initView()
    }
    constructor(context: Context, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) : super(context, attrs, defStyleAttr, defStyleRes) {
        initView()
    }

    private fun initView () {
        var view = LayoutInflater.from(context).inflate(R.layout.layout_pull_to_refreash, this, true)
    }

    override fun onRefreshBefore(scrollY: Int, headerHeight: Int) {

    }

    override fun onRefreshAfter(scrollY: Int, headerHeight: Int) {

    }

    override fun onRefreshReady(scrollY: Int, headerHeight: Int) {

    }

    override fun onRefreshing(scrollY: Int, headerHeight: Int) {

    }

    override fun onRefreshComplete(scrollY: Int, headerHeight: Int, isRefreshSuccess: Boolean) {

    }

    override fun onRefreshCancel(scrollY: Int, headerHeight: Int) {

    }

    override fun getRefreshHeight(): Int {
        return 0
    }
}